clustered/client/src/main/java/org/ehcache/clustered/client/config/TimeoutDuration.java

```diff
@@ -0,0 +1,159 @@
+   * @throws IllegalArgumentException if {@code amount} is negative
+   */
+  public static TimeoutDuration of(long amount, TimeUnit unit) {
+    if (unit == null) {
+      throw new NullPointerException("TimeoutDuration unit can not be null");
+    }
+    if (amount < 0) {
+      throw new IllegalArgumentException("TimeoutDuration time amount must be non-negative");
+    }
+    if (amount == NONE.amount && unit == NONE.unit) {
+      return NONE;
+    }
+    return new TimeoutDuration(amount, unit);
+  }
+
+  /**
+   * Converts this {@code TimeoutDuration} to nanoseconds.  Values are converted
+   * according to the rules for {@code java.util.concurrent.TimeUnit}.
+   *
+   * @return the number of nanoseconds represented by this {@code TimeDuration}
+   */
+  public long toNanos() {
+    return unit.toNanos(amount);
+  }
+
+  /**
+   * Converts this {@code TimeoutDuration} to milliseconds.  Values are converted
+   * according to the rules for {@code java.util.concurrent.TimeUnit}.
+   *
+   * @return the number of milliseconds represented by this {@code TimeDuration}
+   */
+  public long toMillis() {
+    return unit.toMillis(amount);
+  }
+
+  /**
+   * Converts this {@code TimeDuration} to the specified time unit.  Values are
+   * converted according to the rules for {@code java.util.concurrent.TimeUnit}.
+   *
+   * @param toUnit the {@code TimeUnit} to which this {@code TimeDuration} value is converted
+   *
+   * @return the duration expressed in {@code toUnit} units
+   *
+   * @see TimeUnit#convert(long, TimeUnit)
+   */
+  public long convert(TimeUnit toUnit) {
+    return toUnit.convert(amount, unit);
+  }
+
+  /**
+   * Performs a timed wait on the object provided.
+   *
+   * @param obj the {@code Object} on which to wait
+   *
+   * @throws InterruptedException if the wait is interrupted
+   *
+   * @see TimeUnit#timedWait(Object, long)
+   */
+  public void timedWait(Object obj) throws InterruptedException {
+    unit.timedWait(obj, amount);
+  }
+
+  @Override
+  public boolean equals(Object other) {
+    if (this == other) {
+      return true;
+    }
+
+    if (other == null || this.getClass() != other.getClass()) {
+      return false;
+    }
+
+    TimeoutDuration that = (TimeoutDuration)other;
+    if (this.amount == 0 && that.amount == 0) {
+      return true;          // A duration of 0 is the same in any units
+    } else if (this.unit == that.unit) {
+      return this.amount == that.amount;
+    } else {
+      /*
+       * Select the more precise/granular unit for the basis of the equals comparison. Since
+       * TimeUnit.convert _truncates_ fractional values, attempting to convert one a more
+       * precise/granular unit to a less precise/granular unit will result in zero -- indicating
+       * which is the more precise/granular unit.
+       */
+      TimeoutDuration lessGranular;
+      TimeoutDuration moreGranular;
+      if (this.unit.convert(1, that.unit) > 0) {
```

Comment:
Enums are comparable... and TimeUnit is sensibly ordered... so you can just compare the enum instances directly.